Extend the proposal to contain two methods (#4)
rmrk-team/SoubloundEIP: README.md

---
eip: 6454
title: Minimalistic Transferable Non-Fungible Tokens

> description: A minimal interface for Transferable Non-Fungible Tokens extension allowing for tokens to be transferable or non-transferable.

Reviewer: How  about:
```suggestion
description: A minimal extension to identify Transferability on Non-Fungible Tokens.
```

## Abstract

> The Minimalistic Transferable interface for Non-Fungible Tokens standard extends [ERC-721](./eip-721.md) by introducing the ability to identify whether an NFTs can be transfered or not.

Reviewer: ```suggestion
The Minimalistic Transferable interface for Non-Fungible Tokens standard extends [ERC-721](./eip-721.md) by introducing the ability to identify whether NFTs can be transferred or not.
```

Reviewer: No issues. I just wanted to fix singular article with plural noun.

interface IERC6454 /* is IERC165 */ {
    /**
     * @notice Used to check whether the given token is transferable or not.
     * @dev If this function returns `false`, the transfer of the token MUST revert execution
     * @dev If the tokenId does not exist, this method MUST revert execution
     * @param tokenId ID of the token being checked
     * @return Boolean value indicating whether the given token is transferable
     */
    function isTransferable(uint256 tokenId) external view returns (bool);

    /**
     * @notice Used to check whether the given token is transferable or not based on source and destination address.
     * @dev If this function returns `false`, the transfer of the token MUST revert execution

>      * @dev If the tokenId does not exist, this method MUST revert execution

Reviewer: I'd add this to the other one as well

Author: Which one? Both `isTransferable` functions have these `@dev` tags.

2. **Why is there no event marking the token as Non-Transferable in this interface?**\
   The token can become non-transferable either at its creation, after being marked as non-transferable, or after a certain condition is met. This means that some cases of tokens becoming non-transferable cannot emit an event, such as if the token becoming non-transferable is determined by a block number. Requiring an event to be emitted upon the token becoming non-transferable is not feasible in such cases.
3. **Should the transferability state management function be included in this proposal?**\
   A function that marks a token as non-transferable or releases the binding is referred to as the transferability management function. To maintain the objective of designing an agnostic minimal transferable proposal, we have decided not to specify the transferability management function. This allows for a variety of custom implementations that require the tokens to be non-transferable.
4. **Why should this be an EIP if it only contains two methods?**\

>    One could argue that since the core of this proposal is to only prevent ERC-721 tokens to be transferred, this could be done by overriding the transfer function. While this is true, the only way to assure that the token is non-transferable before the smart contract execution, is for it to have the transferable interface.\

Reviewer: Can we mention something about UX?

Author: Good call! I'll add it in the evening!

   This also allows for smart contract to validate whether the token is not transferable and not attempt transferring it as this would result in failed transactions and wasted gas.
5. **Why does this proposal contain two methods with the same name?**\

>    Both methods defined in this proposal address the same issue, but in different ways. The first method is used to check whether the token is transferable or notin general, while the second method is used to check whether the token is conditionally transferable or not based on the source and destination addresses. The second method is useful in cases where the transferability of the token is dependent on the source and destination addresses.

Reviewer: ```suggestion
   Both methods defined in this proposal address the same issue, but in different ways. The first method is used to check whether the token is transferable or not in general, while the second method is used to check whether the token is conditionally transferable or not based on the source and destination addresses. The second method is useful in cases where the transferability of the token is dependent on the source and destination addresses.
```
